File: flut/.flutter/lib/flutter/painting/circle_border.dart
import 'package:flutter/material.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';
import 'package:flut/flutter/painting/borders.dart';

class FlutCircleBorder extends FlutValueObject {
  final CircleBorder value;
  const FlutCircleBorder(this.value) : super('CircleBorder');

  @override
  Map<String, dynamic> flutEncode() {
    final result = flutBaseProps();
    result['side'] = FlutBorderSide(value.side).flutEncode();
    result['eccentricity'] = value.eccentricity;
    return result;
  }

  static CircleBorder? flutDecode(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    return CircleBorder(
      side: runtime.unpackRequiredField<BorderSide>(data, 'side'),
      eccentricity: runtime.unpackRequiredField<double>(data, 'eccentricity'),
    );
  }
}

File: flut/flutter/painting/circle_border.py
from typing import override

from flut._flut.engine import _flut_pack_value
from flut._flut.runtime import _flut_unpack_required_field
from flut.flutter.painting.borders import BorderSide, OutlinedBorder


class CircleBorder(OutlinedBorder):
    _flut_type = "CircleBorder"

    def __init__(
        self,
        *,
        side: BorderSide = BorderSide.none,
        eccentricity: float = 0.0,
    ):
        super().__init__()
        self.side = side
        self.eccentricity = eccentricity

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        result["side"] = _flut_pack_value(self.side)
        result["eccentricity"] = _flut_pack_value(self.eccentricity)
        return result

    @staticmethod
    def _flut_unpack(data: dict) -> "CircleBorder":
        return CircleBorder(
            side=_flut_unpack_required_field(data, "side"),
            eccentricity=_flut_unpack_required_field(data, "eccentricity"),
        )
